zoo/board_games/mcts_bot.py

```diff
@@ -177,52 +237,73 @@ def best_action(self, simulations_number=None, total_simulation_seconds=None):
             Returns the best children node, and can get action from Node.best_action_index.
         """
 
+        # The search cost is determined by either the maximum number of simulations or the longest simulation time.
         if simulations_number is None:
             assert (total_simulation_seconds is not None)
             end_time = time.time() + total_simulation_seconds
             while True:
+                # get the leaf node.
                 v = self._tree_policy()
+                # rollout from the leaf node.
                 reward = v.rollout()
+                # backpropagate from the leaf node to the root node.
                 v.backpropagate(reward)
                 if time.time() > end_time:
                     break
         else:
             for i in range(0, simulations_number):
                 # print('****simlulation-{}****'.format(i))
+                # get the leaf node.
                 v = self._tree_policy()
+                # rollout from the leaf node.
                 reward = v.rollout()
                 # print('reward={}\n'.format(reward))
+                # backpropagate from the leaf node to the root node.
                 v.backpropagate(reward)
         # to select best child go for exploitation only
+        # 这里和算法理论里选择最高访问次数似乎不太一样？？？？？？？？？？？？？？？？？？
         return self.root.best_child(c_param=0.)
 
+    #
     def _tree_policy(self):
         """
```

Comment:
"""
          Implement the tree policy for action selection during rollout.

          At each step, if the current node is not fully expanded, it expands.
          If it is fully expanded, it moves to the best child according to the tree policy.
          This continues until a terminal node is reached, which is then returned.

          Returns:
              The terminal node reached by following the tree policy.
          """